File: core/src/com/g3ida/withflyingcolours/core/player/animation/PlayerAnimationComponent.java
package com.g3ida.withflyingcolours.core.player.animation;

import com.badlogic.gdx.math.Interpolation;

import games.rednblack.editor.renderer.components.BaseComponent;

public class PlayerAnimationComponent implements BaseComponent {

    public TransformAnimation scaleAnimation;
    public TransformAnimation squeezeAnimation;

    public boolean doScale;
    public boolean doSqueeze;

    public static final float SQUEEZE_ANIM_DURATION = 0.17f;
    public static final float SCALE_ANIM_DURATION = 0.17f;

    public PlayerAnimationComponent() {
        super();

        this.scaleAnimation = new TransformAnimation(PlayerAnimationComponent.SCALE_ANIM_DURATION,
                new Interpolation.ElasticIn(1f, 1f, 1, 0.1f),
                true,
                true);

        this.squeezeAnimation = new TransformAnimation(PlayerAnimationComponent.SQUEEZE_ANIM_DURATION,
                new Interpolation.ElasticOut(1f, 1f, 1, 0.1f),
                true,
                false);
        reset();
    }

    @Override
    public void reset() {
        this.doScale = false;
        this.doSqueeze = false;
        this.scaleAnimation.reset(null);
        this.squeezeAnimation.reset(null);
    }
}

File: core/src/com/g3ida/withflyingcolours/core/player/animation/PlayerAnimationComponent.kt
package com.g3ida.withflyingcolours.core.player.animation

import com.artemis.PooledComponent
import com.badlogic.gdx.math.Interpolation

const val SQUEEZE_ANIM_DURATION = 0.17f
const val SCALE_ANIM_DURATION = 0.17f

class PlayerScaleAnimation: TransformAnimation(SCALE_ANIM_DURATION,
    Interpolation.ElasticIn(1f, 1f, 1, 0.1f))
class PlayerSqueezeAnimation: TransformAnimation(SQUEEZE_ANIM_DURATION,
    Interpolation.ElasticOut(1f, 1f, 1, 0.1f), true, false)
class PlayerIdleAnimation: TransformAnimation(0f, Interpolation.linear)

class PlayerAnimationComponent : PooledComponent() {
    var currentAnimation: ITransformAnimation = PlayerIdleAnimation()
    public override fun reset() {
        currentAnimation = PlayerIdleAnimation()
    }
}